dlrover/python/diagnosis/diagnostician/diagnostician.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2025 The DLRover Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+import time
+from typing import Dict
+
+from dlrover.python.common.log import default_logger as logger
+from dlrover.python.diagnosis.common.diagnosis_action import (
+    DiagnosisAction,
+    EventAction,
+    NoAction,
+)
+from dlrover.python.diagnosis.common.diagnosis_observation import (
+    DiagnosisObservation,
+)
+from dlrover.python.util.function_util import TimeoutException, timeout
+
+
+class Diagnostician:
+    """
+    Diagnostician is to observe problems and resolve those problems
```

Comment:
Need more annotation to describe the relationship of 'observe', 'resolve' and 'diagnose'.